src/proto/single/mod.rs

```diff
@@ -71,7 +72,7 @@ pub struct Job {
     ///
     /// This field is read-only.
     #[serde(skip_serializing)]
-    pub(crate) failure: Option<Failure>,
+    pub failure: Option<Failure>,
```

Comment:
I think I made this non-`pub` because it's read-only. It should instead have a getter function that yields `Option<&Failure>` so that users aren't tempted to try to modify it for an outgoing `Job`.

Reply:
Also, I just realized, `Failure`'s fields have no public fields and no accessors, so even if a user _does_ get a `&Failure` there's not much they can do with it. I'd be open to making the fields of `Failure` `pub` though (although with `#[non_exhaustive]`.

Reply:
Makes sense. I'll just revert this for now and potentially revisit in another PR later!